Date: Wed, 06 Apr 2016 22:28:55 +0100
From: Steven Chamberlain <steven@pyro.eu.org>
Subject: Tests: allow newer or equal timestamps in BuildDepends

Allow timestamps to be newer *or equal* in BuildDepends, in case the
filesystem lacks sub-second granularity (such as UFS on FreeBSD).

The test logic is changed from A > B, to !(B > A), i.e. A >= B

--- a/Tests/BuildDepends/CMakeLists.txt
+++ b/Tests/BuildDepends/CMakeLists.txt
@@ -202,12 +202,12 @@
 endif()
 
 if(EXISTS ${BuildDepends_BINARY_DIR}/Project/multi2-real.txt)
-  if(${BuildDepends_BINARY_DIR}/Project/multi2-real.txt
-      IS_NEWER_THAN ${BuildDepends_BINARY_DIR}/Project/multi2-stamp.txt)
-    message(STATUS "multi2-real.txt is newer than multi2-stamp.txt")
-  else()
+  if(${BuildDepends_BINARY_DIR}/Project/multi2-stamp.txt
+      IS_NEWER_THAN ${BuildDepends_BINARY_DIR}/Project/multi2-real.txt)
     message(SEND_ERROR "Project did not initially build properly: "
-      "multi2-real.txt is not newer than multi2-stamp.txt")
+      "multi2-real.txt timestamp is not >= multi2-stamp.txt")
+  else()
+    message(STATUS "multi2-real.txt timestamp is >= multi2-stamp.txt")
   endif()
 else()
   message(SEND_ERROR "Project did not initially build properly: "
@@ -216,12 +216,12 @@
 
 if(TEST_MULTI3)
   if(EXISTS ${BuildDepends_BINARY_DIR}/Project/multi3-real.txt)
-    if(${BuildDepends_BINARY_DIR}/Project/multi3-real.txt
-        IS_NEWER_THAN ${BuildDepends_BINARY_DIR}/Project/multi3-stamp.txt)
-      message(STATUS "multi3-real.txt is newer than multi3-stamp.txt")
-    else()
+    if(${BuildDepends_BINARY_DIR}/Project/multi3-stamp.txt
+        IS_NEWER_THAN ${BuildDepends_BINARY_DIR}/Project/multi3-real.txt)
       message(SEND_ERROR "Project did not initially build properly: "
-        "multi3-real.txt is not newer than multi3-stamp.txt")
+        "multi3-real.txt timestamp is not >= multi3-stamp.txt")
+    else()
+      message(STATUS "multi3-real.txt timestamp is >= multi3-stamp.txt")
     endif()
   else()
     message(SEND_ERROR "Project did not initially build properly: "
@@ -405,12 +405,12 @@
 endif()
 
 if(EXISTS ${BuildDepends_BINARY_DIR}/Project/multi2-real.txt)
-  if(${BuildDepends_BINARY_DIR}/Project/multi2-real.txt
-      IS_NEWER_THAN ${BuildDepends_BINARY_DIR}/Project/multi2-stamp.txt)
-    message(STATUS "multi2-real.txt is newer than multi2-stamp.txt")
-  else()
+  if(${BuildDepends_BINARY_DIR}/Project/multi2-stamp.txt
+      IS_NEWER_THAN ${BuildDepends_BINARY_DIR}/Project/multi2-real.txt)
     message(SEND_ERROR "Project did not rebuild properly: "
-      "multi2-real.txt is not newer than multi2-stamp.txt")
+      "multi2-real.txt timestamp is not >= multi2-stamp.txt")
+  else()
+    message(STATUS "multi2-real.txt is >= multi2-stamp.txt")
   endif()
 else()
   message(SEND_ERROR "Project did not rebuild properly: "
@@ -419,12 +419,12 @@
 
 if(TEST_MULTI3)
   if(EXISTS ${BuildDepends_BINARY_DIR}/Project/multi3-real.txt)
-    if(${BuildDepends_BINARY_DIR}/Project/multi3-real.txt
-        IS_NEWER_THAN ${BuildDepends_BINARY_DIR}/Project/multi3-stamp.txt)
-      message(STATUS "multi3-real.txt is newer than multi3-stamp.txt")
-    else()
+    if(${BuildDepends_BINARY_DIR}/Project/multi3-stamp.txt
+        IS_NEWER_THAN ${BuildDepends_BINARY_DIR}/Project/multi3-real.txt)
       message(SEND_ERROR "Project did not rebuild properly: "
-        "multi3-real.txt is not newer than multi3-stamp.txt")
+        "multi3-real.txt timestamp is not >= multi3-stamp.txt")
+    else()
+      message(STATUS "multi3-real.txt is >= multi3-stamp.txt")
     endif()
   else()
     message(SEND_ERROR "Project did not rebuild properly: "
